domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/casefield/CaseFieldEntityComplexFieldACLValidatorImpl.java

```diff
@@ -0,0 +1,109 @@
+package uk.gov.hmcts.ccd.definition.store.domain.validation.casefield;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.definition.store.domain.validation.ValidationResult;
+import uk.gov.hmcts.ccd.definition.store.domain.validation.authorization.AuthorisationCaseFieldValidationContext;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.CaseFieldACLEntity;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.CaseFieldEntity;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.ComplexFieldACLEntity;
+
+@Component
+public class CaseFieldEntityComplexFieldACLValidatorImpl implements CaseFieldEntityValidator {
+
+    private static final Logger LOG = LoggerFactory.getLogger(CaseFieldEntityComplexFieldsValidatorImpl.class);
+
+    @Override
+    public ValidationResult validate(final CaseFieldEntity caseField,
+                                     final CaseFieldEntityValidationContext caseFieldEntityValidationContext) {
+        final ValidationResult validationResult = new ValidationResult();
+
+        for (ComplexFieldACLEntity entity : caseField.getComplexFieldACLEntities()) {
+            validateUserRole(caseField, caseFieldEntityValidationContext, validationResult, entity);
+            validateCRUDAgainstCaseFieldParent(caseField, caseFieldEntityValidationContext, validationResult, entity);
+            validateCRUDComplexParent(caseField, caseFieldEntityValidationContext, validationResult, entity);
+        }
+        return validationResult;
+    }
+
+    private void validateCRUDComplexParent(CaseFieldEntity caseField, CaseFieldEntityValidationContext caseFieldEntityValidationContext,
+                                           ValidationResult validationResult, ComplexFieldACLEntity complexFieldACLEntity) {
+        String userRole = complexFieldACLEntity.getUserRole() != null ? complexFieldACLEntity.getUserRole().getReference() : "";
+        caseField.getComplexFieldACLEntities()
+            .stream()
+            .anyMatch(entity -> {
+                boolean match = (entity.getUserRole() != null && entity.getUserRole().getReference().equalsIgnoreCase(userRole))
+                    && entity.getListElementCode().contains(complexFieldACLEntity.getListElementCode())
+                    && !entity.getListElementCode().equalsIgnoreCase(complexFieldACLEntity.getListElementCode())
```

Comment:
Not sure it is clear for me from the conditions what are we checking here - could we extract to method with clear names so that it is more readable?
Talking about:
`entity.getListElementCode().contains(complexFieldACLEntity.getListElementCode())`
and
`!entity.getListElementCode().equalsIgnoreCase(complexFieldACLEntity.getListElementCode())`

Reply:
Ok as discussed it is clear now but ant chance we can try and extract these to some self descriptive method names?

Reply:
done 👍